astropy/cosmology/_src/traits/curvature.py

```diff
@@ -0,0 +1,76 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Critical density.
+
+This is private API. See `~astropy.cosmology.traits` for public API.
+
+"""
+
+__all__ = ["CurvatureComponent"]
+
+__doctest_requires__ = {"CurvatureComponent.Ok0": ["numpy>=2.0"]}
+
+from typing import Any
+
+import numpy as np
+import numpy.typing as npt
+from numpy.typing import ArrayLike
+
+from astropy.cosmology._src.utils import aszarr, deprecated_keywords
+from astropy.units import Quantity
+
+
+class CurvatureComponent:
+    """The object has attributes and methods related to the global curvature."""
+
+    Ok0: float | np.floating[Any]
+    """Omega curvature; the effective curvature density/critical density at z=0."""
+
+    @deprecated_keywords("z", since="7.0")
+    def Ok(self, z: Quantity | ArrayLike) -> npt.NDArray | np.floating[Any]:
+        """Return the equivalent density parameter for curvature at redshift ``z``.
+
+        Parameters
+        ----------
+        z : Quantity-like ['redshift'], array-like
+            Input redshift.
+
+            .. versionchanged:: 7.0
+                Passing z as a keyword argument is deprecated.
+
+        Returns
+        -------
+        Ok : ndarray
+            The equivalent density parameter for curvature at each redshift.
+
+            .. versionchanged:: 7.1
+                Always returns a numpy object, never a `float`.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from astropy.cosmology import Planck18, units as cu
+
+        >>> Planck18.Ok(2)
+        array(0.)
+
+        >>> Planck18.Ok([1, 2])
+        array([0., 0.])
+
+        >>> Planck18.Ok(np.array([2]))
+        array([0.])
+
+        >>> Planck18.Ok(2 * cu.redshift)
+        array(0.)
+
+        >>> cosmo = Planck18.clone(Ode0=0.71, to_nonflat=True)
+        >>> cosmo.Ok0
+        np.float64(-0.021153694455455927)
```

Comment:
I must admit I don't know what to think of these examples. I'll just have to assume that if the tests succeed then the examples work. But this example does not.

Reply:
That's because numpy changed how stuff is printed. This succeeds on numpy v2.0+. My `__doctest_requires__` isn't working.

Reply:
@eerovaher @mhvk have you encountered this / know a fix?

Reply:
Didn't work and I think can't work b/c this docstring is inherited.

Reply:
We set up the skipping for these kinds of numpy format problems in `pyproject.toml` -- https://github.com/astropy/astropy/blob/27a861e2746d9411999b45d521cfd21e94015ac6/pyproject.toml#L239-L261